client/packages/inventory/src/Stocktake/DetailView/ContentArea/columns.ts

```diff
@@ -151,6 +165,46 @@ export const useStocktakeColumns = ({
           },
         },
       ],
+      [
+        'locationName',
+        {
+          getSortValue: row => {
+            if ('lines' in row) {
+              const locations = row.lines
+                .map(({ location }) => location)
+                .filter(Boolean) as LocationRowFragment[];
+              if (locations.length !== 0) {
+                return ArrayUtils.ifTheSameElseDefault(
+                  locations,
+                  'name',
+                  `${t('multiple')}`
+                );
+              } else {
+                return '';
+              }
+            } else {
+              return row.location?.name ?? '';
+            }
+          },
+          accessor: ({ rowData }) => {
+            if ('lines' in rowData) {
+              const locations = rowData.lines
+                .map(({ location }) => location)
+                .filter(Boolean) as LocationRowFragment[];
```

Comment:
"as" is big red flag for me for type safety, it might be ok now but things could break if say something new is added to LocationRowFragment but not to this [query](https://github.com/openmsupply/open-msupply/blob/508b6e5a9ac46f08b7bac9a5f9276d1f052faa39/client/packages/system/src/Item/api/operations.graphql#L16-L21), etc..

You can use flatMap here:

const locations = rowData.lines.flatMap(({location}) => !!location ? [location] : []);

Reply:
[0a8687a](https://github.com/openmsupply/open-msupply/pull/2302/commits/0a8687a545df503a9218a2536e6e789c69e9ed2e)